### PR TITLE
Add support for _headers files

### PR DESCRIPTION
**Pull request by RaeesBhatti**
**- Summary**

Fixes: https://github.com/netlify/cli/issues/446

**- Test plan**
Have the following file in your project root:
```
# a path:
/templates/index.html
  # headers for that path:
  X-Frame-Options: DENY
  X-XSS-Protection: 1; mode=block
# another path:
/templates/index2.html
  # headers for that path:
  X-Frame-Options: SAMEORIGIN
/*
  cache-control: max-age=0
  cache-control: no-cache
  cache-control: no-store
  cache-control: must-revalidate
```
Run `netlify dev`, make an http request to `/` and observe response headers.

**- Description for the changelog**
* Add new function `parseHeadersFile`
* Watch for changes in `_headers` files in project root and dist directory
* Set headers defined in `_headers` files to responses

**- A picture of a cute animal (not mandatory but encouraged)**
🦁

### CODE REVIEW

**Comment by RaeesBhatti** [comment]
Just realized that this is missing support for `*` and `:placeholder`.

**Comment by RaeesBhatti** [comment]
Specification at https://docs.netlify.com/routing/headers/

File: src/commands/dev/index.js [hunk 1]
```diff
@@ -97,6 +99,17 @@ function initializeProxy(port) {
...
 
+  const headersFiles = Array.from(new Set([
+    path.resolve(projectDir, '_headers'),
+    path.resolve(distDir, '_headers'),
+  ]))
+
+  let headerRules = {}
+  onChanges(headersFiles, () => {
+    console.log(`${NETLIFYDEVLOG} Reloading headers files`, headersFiles.map(p => path.relative(projectDir, p)))
+    headerRules = headersFiles.reduce((prev, curr) => Object.assign(prev, parseHeadersFile(curr)), {})
```
Comment (erezrokah) [inline src/commands/dev/index.js]: Should we catch `parseHeadersFile` errors in this callback and log them?

Comment (erezrokah) [inline src/commands/dev/index.js]: Should we catch `parseHeadersFile` errors in this callback and log them?

Comment (RaeesBhatti) [reply src/commands/dev/index.js]: We're throwing those errors without catching because we want to halt the execution if a rule rule is not correct.

**Comment by erezrokah** [comment]
With this PR merged the cli fails on `gatsby-starter-netlify-cms`.
Looks like the header file in that repo is not in the expected format:
https://github.com/netlify-templates/gatsby-starter-netlify-cms/blob/9d00ee28c82519f331eceff3edfdfabdff051ff9/_headers

Deploy succeeds though and shows 41 header rules processed. 
After the CLI fails, Gatsby server remains running in the background.
Also a browser window is still opened regardless of the failure.

@RaeesBhatti thoughts?

**Comment by RaeesBhatti** [comment]
The headers file you provided doesn't look correct according to the specification: https://docs.netlify.com/routing/headers/#syntax-for-the-headers-file. The syntax used there is of a TOML file. I would suggest using the correct syntax for `_headers` or putting those rules in `netlify.toml` instead.
I think the production is being more lenient on this by trying to parse it as TOML if _headers fails. IMO we shouldn't implement that behavior in the CLI (not right now at least) since the syntax for an `_headers` file is clearly specified in the docs.

**Comment by RaeesBhatti** [comment]
As for the CLI failing because of syntax errors and Gatsby server staying alive is an issue that we should definitely look at.

**Comment by erquhart** [comment]
@rybit it looks like we allow `_headers` files to have TOML content, but this behavior isn't documented. I'm guessing this is a backwards compatibility choice - can you confirm, or mention someone who can?

@RaeesBhatti if this is indeed something we're intentionally supporting in the platform, we should mirror it in the CLI. But let's see where this discussion goes.

**Pull request merged**